storage-account-rd-data-extracts.tf

```
@@ -2,6 +2,21 @@ locals {
   rd_data_extract_product         = "rddataextract"
   rd_data_extract_account_name    = join("", [local.rd_data_extract_product, var.env])
   rd_data_extract_container_name  = "rd-data-extract"
+
+  allowed_roles = [
+    "Storage Account Delegator",
```

Storage Account Delegator isn't a role: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles
```suggestion
    "Storage Account Delegator",
```